js/app.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+var productLineUp = [];
+
+
+function Product (name, path) {
+  this.name = name;
+  this.path = path;
+  this.clicks = 0;
+  this.timesShown = 0;
+  productLineUp.push(this);
+}
+
+productLineUp.getElWithId = function(id) {
+  for (var ii = 0; ii < productLineUp.length; ii++) {
+    var obj = productLineUp[ii];
+    if (obj.name === id) {
+      return obj;
+    }
+  }
+  console.log('Could not find id');
+};
+
+function clickCounter (event) {
+  var idName = event.target.getAttribute('id');
+  var object = productLineUp.getElWithId(idName);
+  object.clicks++;
+  console.log(object);
+}
+
+var clickPic = document.getElementById('pic');
+clickPic.addEventListener('click', clickCounter);
+
+
+
+// var currentPics = [];
+function random (array) {
+  var temp = Math.floor(Math.random() * productLineUp.length);
+  while (array.includes(temp)) {
+    temp = Math.floor(Math.random() * productLineUp.length);
+  }
+  return temp;
+}
+
+
+function renderPic () {
+  var indexed = [];
+  for (var i = 0; i < 3; i++) {
+    var tempRandom = random(indexed);
+    indexed.push(tempRandom);
+    var parentEl = document.getElementById('pic');
+    var fig = document.createElement('figure');
+    var figcaption = document.createElement('figcaption');
+    fig.setAttribute('id', 'fig' + i);
+    figcaption.textContent = productLineUp[tempRandom].name;
+    var img = document.createElement('img');
+    img.setAttribute('src', 'images/' + productLineUp[tempRandom].path);
+    img.setAttribute('id', productLineUp[tempRandom].name);
+    parentEl.appendChild(img);
+    parentEl.appendChild(fig);
+    fig.appendChild(figcaption);
+  }
+}
```

Comment:
You probably want to have an internal index array and then you can update the outside index before the function exist. Something like `outsideIndex = insideIndexed `